### PR TITLE
ENT-11134: proper message when invalid cloud credentials

### PR DESCRIPTION
**Pull request by mineralsfree**
New invalid credentials error message: 
image

Ticket: ENT-11134
Changelog: None

### CODE REVIEW

File: cf_remote/commands.py [hunk 1]
```diff
@@ -403,6 +403,9 @@ def spawn(
         role=role,
         spawned_cb=print_progress_dot,
     )
+    if len(vms) == 0:
```
Comment (larsewi) [inline cf_remote/commands.py:406]: You could do
```suggestion
    if vms:
```
since empty list evaluates to `False`, but maybe your way is more readable 🤔

Comment (larsewi) [reply cf_remote/commands.py:406]:
```
>>> if not []:
...     print("Empty")
... else:
...     print("Not empty")
... 
Empty
```

File: cf_remote/spawn.py [hunk 2]
```diff
@@ -457,9 +460,10 @@ def spawn_vms(
                 req.size,
                 role,
             )
-            if spawned_cb is not None:
-                spawned_cb(vm)
-            ret.append(vm)
+            if vm is not None:
+                ret.append(vm)
+                if spawned_cb is not None:
+                    spawned_cb()
```
[the remark this answers is not in the document]
Comment (mineralsfree) [reply cf_remote/spawn.py:466]: mistake :sweat_smile:

File: cf_remote/commands.py (hunk 1 above)
Comment (larsewi) [reply cf_remote/commands.py:406]: I think maybe your way is clearer 😉

Comment (mineralsfree) [reply cf_remote/commands.py:406]: :blush:

**Review by larsewi — APPROVED** [review]

File: cf_remote/spawn.py [hunk 3]
```diff
@@ -307,9 +308,12 @@ def spawn_vm_in_aws(
...
 ):
-    driver = get_cloud_driver(Providers.AWS, aws_creds, region)
-    existing_vms = driver.list_nodes()
-
+    try:
+        driver = get_cloud_driver(Providers.AWS, aws_creds, region)
+        existing_vms = driver.list_nodes()
+    except InvalidCredsError as error:
+        print("\nInvalid credentials, check cloud_config.json file, details: \n" + str(error))
+        return None
```
Comment (olehermanse) [inline cf_remote/spawn.py:316]: Same as in other PR. This function already uses `raise ValueError` to handle erors. It seems to me you should also use that here, unless you have a good reason not to.

**Review by olehermanse — CHANGES_REQUESTED** [review]

File: cf_remote/commands.py (hunk 1 above)
Comment (olehermanse) [reply cf_remote/commands.py:406]: You've updated the function to return None, this won't work;

```
$ python3
Python 3.11.6 (main, Oct  2 2023, 20:14:46) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> len(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'NoneType' has no len()
```

File: cf_remote/spawn.py (hunk 3 above)
Comment (olehermanse) [reply cf_remote/spawn.py:316]: no; discussed the better way to do this in person.

**Review by olehermanse — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**